### DISCUSSION TITLE
Internal server error

### DISCUSSION THREAD

**Discussion by ghost** [searxng/searxng#1517] [Q&A]
So I have installed searxng to the host (no Docker) and it connects regularly to localhost:8888. Then I edited the settings.yml file in order to use Tor, adding the section:

> outgoing:
request_timeout: 2.0       # default timeout in seconds, can be override by engine
  max_request_timeout: 10.0  # the maximum timeout in seconds
  useragent_suffix: ""       # informations like an email address to the administrator
  pool_connections: 100      # Maximum number of allowable connections, or null
                             # for no limits. The default is 100.
  pool_maxsize: 10           # Number of allowable keep-alive connections, or null
                             # to always allow. The default is 10.
  enable_http2: true         # See https://www.python-httpx.org/http2/

>   proxies: 
all://: socks5h:127.0.0.1:9050
using_tor_proxy=true

But after saving, I get a internal server error.

**Comment by return42** [comment]
> Then I edited the settings.yml 

The settings.yml file is a [YAML](https://yaml.org/spec/1.1/) file.  Most often the indentation of the modified lines is not OK, please check your yml file and give it a next try.

**Reply by ghost** [reply]
Sorry, when I pasted the code the indentation was gone 😁I think it is correct in the file because now the behavior is erratic, I mean sometimes it surfs, at the following reboot or logout the error may comes back.

**Reply by ghost** [reply]
It turned out to be a misconfigured uwsgi service. Problem (tentatively) fixed.

**Reply by mrpaulblack** [reply]
So it worked after you changed our `uwsgi.ini` file? Can you post your changes here please? Or was it something unique to your system?

**Reply by ghost** [reply]
Hey Paul, yes of course 😊it's something in between, my system doesn't have systemd, so I had to start uwsgi with a dedicated script. The exec command is `uwsgi - - master - - die-on-term - - emperor /etc/uwsgi`. Also, I delete the line logger = systemd in searxng.ini. The only problem is that shutting down the system has become very sluggish, maybe I should find a better way to stop the uwsgi server.
